lib/smbios.c

```diff
@@ -515,6 +522,47 @@ static int smbios_write_type4(ulong *current, int handle,
 	return len;
 }
 
+static int smbios_write_type7_level(ulong *current, int handle,
+				    struct smbios_ctx *ctx, int level)
+{
+	struct smbios_type7 *t;
+	int len = sizeof(struct smbios_type7);
+
+	t = map_sysmem(*current, len);
+	memset(t, 0, sizeof(struct smbios_type7));
+	fill_smbios_header(t, SMBIOS_CACHE_INFORMATION, len, handle);
+	smbios_set_eos(ctx, t->eos);
+
+	printf("=====================> write type7 via sysinfo_get_cache_info\n");
+
+	sysinfo_get_cache_info(level, t);
+
+	len = t->length + smbios_string_table_len(ctx);
+	*current += len;
+	unmap_sysmem(t);
+
+	return len;
+}
+
+static int smbios_write_type7(ulong *current, int handle,
+			      struct smbios_ctx *ctx)
+{
+	int level = 0; /* L1 by default*/
```

Comment:
Does the spec give any guidance? Or should we eventually print L1, L2, L3?

Reply:
The spec only require to have one stucture for each cache, but not require to print L1, L2 and L3.